transports/tls/src/upgrade.rs

```diff
@@ -121,13 +121,9 @@ where
 fn extract_single_certificate(
     state: &CommonState,
 ) -> Result<P2pCertificate<'_>, certificate::ParseError> {
-    let cert = match state
-        .peer_certificates()
-        .expect("config enforces presence of certificates")
-    {
-        [single] => single,
-        _ => panic!("config enforces exactly one certificate"),
-    };
-
-    certificate::parse(cert)
+    if let Some([cert]) = state.peer_certificates() {
```

Comment:
Why use `if let` here? Can't we pattern-match to `Some([cert])` and then call `certificate::parse` in the "happy path" further down?